src/tensorwaves/interfaces.py

```diff
@@ -111,27 +119,56 @@ class Estimator(ABC):
     """Estimator for discrepancy model and data."""
 
     @abstractmethod
-    def __call__(
-        self, parameters: Mapping[str, Union[float, complex]]
-    ) -> float:
+    def __call__(self, parameters: Mapping[str, ParameterValue]) -> float:
         """Evaluate discrepancy."""
 
     @abstractmethod
     def gradient(
-        self, parameters: Mapping[str, Union[float, complex]]
-    ) -> Dict[str, Union[float, complex]]:
+        self, parameters: Mapping[str, ParameterValue]
+    ) -> Dict[str, ParameterValue]:
         """Calculate gradient for given parameter mapping."""
 
 
+@attr.s(frozen=True, auto_attribs=True)
+class FitResult:  # pylint: disable=too-many-instance-attributes
+    minimum_valid: bool
+    execution_time: float
+    function_calls: int
+    estimator_value: float
+    parameter_values: Dict[str, ParameterValue]
+    parameter_errors: Optional[Dict[str, ParameterValue]] = None
+    iterations: Optional[int] = None
+    specifics: Optional[Any] = None
```

Comment:
I would either do one of these (both has its pros and cons):

1. not creating any class, and just return the optimizer/minimizer specific objects directly (for minuit a custom object would have to be created as it does not define one)
2. make the specifics into a dictionary of optional info

Reply:
Yes this is true. I would not favour 1, because it leaves the interface undefined (even though, internally, the package currently has no need for a `FitResult` interface).